### PR TITLE
Reorder saliency maps for h-cls according to labels in .xml meta

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
# What does this PR do?

Only Python part fixed. #TODO CPP part
Reorder saliency maps for h-cls according to labels in .xml meta

Fixes # CVS-132079

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by GalyaZalesskaya** [comment]
@sovrasov  Please review Python part

File: model_api/python/openvino/model_api/models/classification.py [hunk 1]
```diff
@@ -170,11 +171,23 @@ def postprocess(self, outputs, meta):
 
         return ClassificationResult(
             result,
-            outputs.get(_saliency_map_name, np.ndarray(0)),
+            self.get_saliency_maps(outputs),
             outputs.get(_feature_vector_name, np.ndarray(0)),
             raw_scores,
         )
 
+    def get_saliency_maps(self, outputs: Dict) -> np.ndarray:
```
Comment (sovrasov) [inline model_api/python/openvino/model_api/models/classification.py:179]: A short docstring would be useful here

File: model_api/python/openvino/model_api/models/classification.py [hunk 2]
```diff
@@ -170,11 +171,23 @@ def postprocess(self, outputs, meta):
...
+        saliency_maps = outputs.get(_saliency_map_name, np.ndarray(0))
+        if not self.hierarchical:
+            return saliency_maps
+        # In hierarchical case reorder saliency maps to match the order of labels in .XML meta
```
Comment (sovrasov) [inline model_api/python/openvino/model_api/models/classification.py:183]: This can be moved to a docstring

Comment (GalyaZalesskaya) [reply model_api/python/openvino/model_api/models/classification.py]: Thank you, fixed

**Review by sovrasov — APPROVED** [review]

**Pull request merged**